### PR TITLE
Fixed support for Integrators and ServiceContributors in Hibernate ORM

### LINKED ISSUE #7189: Function integrate of custom hibernate integrator is not called
**Describe the bug**
We need to use hibernate integrators to put events listener. And it seems that the custom integrator is only instantiated but never really called. 

**Expected behavior**
I want to add custom integrator and the function integrate is called.

**Actual behavior**
We tried to add the file `META-INF/services/org.hibernate.integrator.spi.Integrator` with the following content : 
```
com.decathlon.samourai.recorder.config.ReplicationEventListenerIntegrator
```

The integrator is added by hibernate as the log suggested : 
```
2020-02-13 16:07:20,631 DEBUG [org.hib.int.int.IntegratorServiceImpl] (main) Adding Integrator [com.decathlon.samourai.recorder.config.ReplicationEventListenerIntegrator].
```
[ReplicationEventListenerIntegrator code](https://gist.github.com/kratisto/3d1eeef1bc101feb3f8f3daf41d422d9)
When we set a debug point on the integrate function, we doesn't go in it. The listener are not set. 
We try to log in the [RecordEventListener](https://gist.github.com/kratisto/4f2070528e255494c7d085a9b03673df) but didn't see anything in the console. And the breakpoint on the log is not triggered. 

**A possible solution**

I look at the code of the `quarkus-hibernate-orm`extensions and make some modification. With theses it work. 

I first add the config in the HibernateOrmConfig
```
    /**
     * List of the integrator to set up. Comma separeted. Fully class name
     */
    @ConfigItem
    public Optional<List<String>> integrators;
```
Then i add in the HibernateOrmProcessor class at line 227 : 
```
        // add integrator present in configuration
        hibernateConfig.integrators.ifPresent(integrators -> {
            for (String integrator : integrators) {
                integratorClasses.add((Class<? extends Integrator>) recorderContext.classProxy(integrator));
            }
        });
```

I modify my `application.properties` with : 
```
quarkus.hibernate-orm.integrators=com.decathlon.samourai.recorder.config.ReplicationEventListenerIntegrator
```

After that all the breakpoints are triggered and i can see the log of the eventlistener. 

I haven't tested natively if it works with or without the patch. 

I also don't know if this fix is the best way to solve the problem. 
It has the advantage of allowing custom integrator configuration.

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
```
Darwin FRL-C02P2645G3QC 18.7.0 Darwin Kernel Version 18.7.0: Tue Aug 20 16:57:14 PDT 2019; root:xnu-4903.271.2~2/RELEASE_X86_64 x86_64
```
 - Output of `java -version`: 
```
openjdk version "11.0.2" 2018-10-16
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.2+7)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.2+7, mixed mode)
```
 - Quarkus version or git rev: 
```
1.2.0.Final
```

### PR DESCRIPTION
**Pull request by gastaldi**
Fixes #7189

### CODE REVIEW

**Comment by gastaldi** [comment]
I also added the service files to be hot-reloaded (and they work like a charm!) :)

**Comment by gastaldi** [comment]
It looks like with #7162 the  `EnversIntegrator` and `EnversServiceContributor`  are no longer automatically registered.

**Comment by gastaldi** [comment]
I deleted my previous comment about Hibernate Envers, the project needs to add the `hibernate-envers` JARs to load it (I thought it was embedded in Hibernate). This PR is good to go :+1:

**Comment by geoand** [comment]
Seems reasonable to me, but I'm going to leave it to the experts 😉

File: extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/JPAConfig.java [hunk 1]
```diff
@@ -74,7 +75,7 @@ boolean isJtaEnabled() {
      *
      * @param event ignored
      */
-    void destroy(@BeforeDestroyed(ApplicationScoped.class) Object event) {
+    void destroy(@Observes @BeforeDestroyed(ApplicationScoped.class) Object event) {
```
Comment (Sanne) [inline extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/JPAConfig.java]: So this wasn't being called?

**Comment by Sanne** [comment]
looks good :+1: many thanks @gastaldi ! And I appreaciate the thourough integration test.

Just have a dumb question regarding the observer?

File: extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/JPAConfig.java (hunk 1 above)
Comment (gastaldi) [reply extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/JPAConfig.java]: It wasn't. That's what was missing to have `Integrator#disintegrate` being called :wink:

Comment (Sanne) [reply extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/JPAConfig.java]: ic. Maybe you have time to add an integration test for `#disintegrate` as well?

**Review by Sanne — APPROVED** [review]

**Comment by gsmet** [comment]
For the record, I backported 6d1eb22d2bbe8617276e20a26477deb99e1f231c to 1.2.1.Final.

**Pull request merged**